packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/useChartCartesianAxis.types.ts

```diff
@@ -47,6 +48,19 @@ export interface UseChartCartesianAxisParameters<S extends ScaleName = ScaleName
    * @param {null | ChartsAxisData} data The data about the clicked axis and items associated with it.
    */
   onAxisClick?: (event: MouseEvent, data: null | ChartsAxisData) => void;
+  /**
+   * The function called when the pointer position corresponds to a new axis data item.
+   * This update can either be caused by a pointer movement, or an axis update.
+   * In case of multiple axes, the function get called if at least ones axis is updated.
+   * And the parameter contains the identifier for all axes with a `data` property.
+   * @param {AxisItemIdentifier[]} newAxisItems The array of axes item identifiers.
+   */
+  onAxisInteraction?: (newAxisItems: AxisItemIdentifier[]) => void;
```

Comment:
I'd call them `axisItems` since they aren't "new". If there are multiple axes, some might be the same as the previous interaction. 

```suggestion
  onAxisInteraction?: (axisItems: AxisItemIdentifier[]) => void;
```